src/collectors/mod.rs

```diff
@@ -1,18 +1,46 @@
 mod github_rate_limit;
+mod github_runners;
 
 pub use crate::collectors::github_rate_limit::GitHubRateLimit;
+pub use crate::collectors::github_runners::GithubRunners;
 
 use crate::MetricProvider;
 use anyhow::{Error, Result};
 use futures::TryFutureExt;
 use log::info;
+use reqwest::header::{HeaderMap, ACCEPT, AUTHORIZATION, USER_AGENT};
+use reqwest::Client;
 
 // register collectors for metrics gathering
 pub async fn register_collectors(p: &MetricProvider) -> Result<(), Error> {
+    let http = Client::new();
```

Comment:
A neat thing provided by `Client` is the ability to set default headers for all requests, by doing something like this:

```rust
let http = ClientBuilder::new().default_headers(headers_map).build();
```

Now, I don't think we want to use it for all the default headers here, especially for the GitHub-specific ones. I don't want to accidentally send our GitHub credentials to third parties if we ever make HTTP requests to other domains, so the GitHub-specific ones should be still set per request.

What we can do is set the user agent here though. Since we need to only set the user agent there is actually a nice method to do just that:

```rust
let http = ClientBuilder::new().user_agent("foobar").build();
```

Reply:
done.